### PR TITLE
ROADMAP: Remove utils.stats

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses roadmap.

Removes `utils.stats` from pysat 3.0.0, as the functions within have been migrated to pysatSeasons and scipy 1.4.0. Note that since the removal of deprecated functions occurs in 3.0.0 and not before, there is no twin to this pull request.  

Also updates the `omni_hro` instrument to use the `circstd` routine migrated to scipy.  Updates requirements in setup.py to note that scipy 1.4.0 is the minimum required. 

## Type of change

- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)

# How Has This Been Tested?

via travis-ci

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
*Note regarding python 2.7 compatibility:  Scipy 1.4.0 is not python 2.7 compatible. This version is required to maintain the handling of NaNs in the `circmean` and `circstd` functions used.  Prior to this version, both functions existed but did not have a `nan_policy`.  The usage of `circstd` in `omni_hro` is not affected in the unit tests, however, as NaNs are not incorporated there.  This is technically a hole in the testing.  (This did not exist before as the special stat functions were tested separately.)  Fixing this hole requires a lapse in python 2.7 compatibility in the unit tests.

Adding to agenda for next meeting.

**Comment by jklenzing** [comment]
Scratch that.  I wasn't passing the `nan_policy` keyword through the function as I should have been.  Once this is in place, python 2.7 tests fail on the two `omni_hro` tests that use this function giving a `TypeError` as expected.

```
TypeError: circstd() got an unexpected keyword argument 'nan_policy'
```

Python 3.7 tests work fine.  Let's discuss at the next meeting.

**Comment by jklenzing** [comment]
TODO: add a NaN to the unit tests to make sure that behaviour of `circstd` remains consistent in future.

**Comment by jklenzing** [comment]
Two errors in python 2.7 related to incompatibility with scipy versions:
```
======================================================================
ERROR: Test the IMF steadiness standard deviation calculation.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/pysat/pysat/pysat/tests/test_omni_hro.py", line 122, in test_clock_angle_std
    min_window_frac=0.8)
  File "/home/travis/build/pysat/pysat/pysat/instruments/omni_hro.py", line 228, in calculate_imf_steadiness
    raw=True)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 1703, in apply
    func, raw=raw, args=args, kwargs=kwargs)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 1012, in apply
    center=False, raw=raw)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 880, in _apply
    result = calc(values)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 874, in calc
    closed=self.closed)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 1009, in f
    closed, offset, func, raw, args, kwargs)
  File "pandas/_libs/window.pyx", line 1638, in pandas._libs.window.roll_generic
TypeError: circstd() got an unexpected keyword argument 'nan_policy'

======================================================================
ERROR: Test the IMF steadiness CV calculation.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/pysat/pysat/pysat/tests/test_omni_hro.py", line 102, in test_yz_plane_cv
    min_window_frac=0.8)
  File "/home/travis/build/pysat/pysat/pysat/instruments/omni_hro.py", line 228, in calculate_imf_steadiness
    raw=True)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 1703, in apply
    func, raw=raw, args=args, kwargs=kwargs)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 1012, in apply
    center=False, raw=raw)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 880, in _apply
    result = calc(values)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 874, in calc
    closed=self.closed)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pandas-0.24.2-py2.7-linux-x86_64.egg/pandas/core/window.py", line 1009, in f
    closed, offset, func, raw, args, kwargs)
  File "pandas/_libs/window.pyx", line 1638, in pandas._libs.window.roll_generic
TypeError: circstd() got an unexpected keyword argument 'nan_policy'
```

**Comment by jklenzing** [comment]
Take a look at the changes so we can discuss next meeting.

**Comment by rstoneback** [comment]
Thanks Jeff. I've taken a look. Ready to discuss on Wed. Cheers.

**Comment by jklenzing** [comment]
@aburrell, any comments on the post-meeting updates?

**Comment by jklenzing** [comment]
> TODO: add a NaN to the unit tests to make sure that behaviour of `circstd` remains consistent in future.

Since we are not forcing the nan_policy, we are not doing this.

**Comment by jklenzing** [comment]
PR fails on DMSP server issue. Push fully successful

File: pysat/instruments/omni_hro.py [hunk 1]
```diff
@@ -199,7 +200,7 @@ def calculate_imf_steadiness(inst, steady_window=15, min_window_frac=0.75,
         Y-Z plane (default=0.5)
     """
 
-    from pysat.utils import stats as pystats
+    import warnings
```
Comment (aburrell) [inline pysat/instruments/omni_hro.py]: All imports should happen up top.

Comment (jklenzing) [reply pysat/instruments/omni_hro.py]: fixed

**Pull request merged**